programs/mango-v4/src/instructions/token_charge_collateral_fees.rs

```diff
@@ -93,7 +113,24 @@ pub fn token_charge_collateral_fees(ctx: Context<TokenChargeCollateralFees>) ->
             continue;
         }
 
-        let fee = token_balance * scaling * I80F48::from_num(bank.collateral_fee_per_day);
+        let token_balance = token_position.native(&bank);
```

Comment:
read the balance from the token_balances array here - position.native() doesn't include reserved serum/obv2 amounts, perp amounts etc